spec/requests/v1/clusters_spec.rb

```diff
@@ -61,35 +74,67 @@
         expect(response.body).to match(/Couldn't find Cluster/)
       end
     end
+
+    context 'when the actor does not have permission' do
+      let(:project_id) { project_unathorised.id }
+      let(:id) { clusters_unathorised.first.id }
+
+      it 'returns status code 403' do
+        expect(response).to have_http_status(403)
+      end
+    end
   end
 
   # Test suite for PUT /projects/:project_id/roles
   describe 'POST /v1/projects/:project_id/clusters' do
-    let(:valid_attributes) { { name: "TestCluster", description: "Stuff", endpoint: "tcp://0.0.0.0:2324", cert: "ssdffsf", key: "sffsdsfd", ca: "adsadsads" } }
+    let(:valid_attributes) do
+      { name: 'TestCluster', description: 'Stuff', endpoint: 'tcp://0.0.0.0:2324', cert: 'ssdffsf', key: 'sffsdsfd', ca: 'adsadsads' }.to_json
```

Comment:
Minor thing: when using dummy strings, I'd suggest getting used to preferring things like "dummy_cert", "dummy_key", to "adsadsads" :p